src/frontends/onnx/frontend/src/op/org.openvinotoolkit/generate_proposals.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "generate_proposals.hpp"
+
+#include "default_opset.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+
+namespace {
+void validate_generate_proposals_inputs(const OutputVector& inputs) {
+    OPENVINO_ASSERT(inputs.size() == 4, "GenerateProposals operator expects 4 inputs, got ", inputs.size());
+
+    const auto scores_rank = inputs[0].get_partial_shape().rank();
+    OPENVINO_ASSERT(scores_rank.compatible(4), "GenerateProposals input scores rank should be 4, is ", scores_rank);
+
+    const auto& anchors_shape = inputs[3].get_partial_shape();
+    const auto anchors_rank = anchors_shape.rank();
+    OPENVINO_ASSERT(anchors_rank.compatible(2), "GenerateProposals input anchors rank should be 2, is ", anchors_rank);
+    OPENVINO_ASSERT(anchors_shape[1].compatible(4),
+                    "GenerateProposals input anchors shape should be {A, 4}, is ",
+                    anchors_shape);
```

Comment:
```suggestion
    if (anchors_rank.is_static()) {
        OPENVINO_ASSERT(anchors_shape[1].compatible(4),
                        "GenerateProposals input anchors shape should be {A, 4}, is ",
                        anchors_shape);
     }
```

Reply:
Yes, it should be static. I'll strengthen the assertion one line above though.